cupyx/texture/_affine.py

```diff
@@ -0,0 +1,159 @@
+import cupy
+
+from cupy.cuda import texture
+from cupy.cuda import runtime
+
+
+def _create_texture_object(data,
+                           address_mode: str,
+                           filter_mode: str,
+                           read_mode: str):
+
+    if cupy.issubdtype(data.dtype, cupy.unsignedinteger):
+        fmt_kind = runtime.cudaChannelFormatKindUnsigned
+    elif cupy.issubdtype(data.dtype, cupy.integer):
+        fmt_kind = runtime.cudaChannelFormatKindSigned
+    elif cupy.issubdtype(data.dtype, cupy.floating):
+        fmt_kind = runtime.cudaChannelFormatKindFloat
+    else:
+        raise ValueError('Unsupported data type')
+
+    if address_mode == 'wrap':
+        address_mode = runtime.cudaAddressModeWrap
+    elif address_mode == 'clamp':
+        address_mode = runtime.cudaAddressModeClamp
+    elif address_mode == 'mirror':
+        address_mode = runtime.cudaAddressModeMirror
+    elif address_mode == 'border':
+        address_mode = runtime.cudaAddressModeBorder
+    else:
+        raise ValueError(
+            'Unsupported address mode '
+            '(supported: wrap, clamp, mirror, border)')
+
+    if filter_mode == 'nearest':
+        filter_mode = runtime.cudaFilterModePoint
+    elif filter_mode == 'linear':
+        filter_mode = runtime.cudaFilterModeLinear
+    else:
+        raise ValueError(
+            'Unsupported filter mode (supported: nearest, linear)')
+
+    if read_mode == 'element_type':
+        read_mode = runtime.cudaReadModeElementType
+    elif read_mode == 'normalized_float':
+        read_mode = runtime.cudaReadModeNormalizedFloat
+    else:
+        raise ValueError(
+            'Unsupported read mode '
+            '(supported: element_type, normalized_float)')
+
+    texture_fmt = texture.ChannelFormatDescriptor(
+        data.itemsize * 8, 0, 0, 0, fmt_kind)
+    # print(f'{data.shape=}, {data.itemsize=}, {data.dtype=}')
+    # CUDAArray: last dimension=fastest changing dimension
+    array = texture.CUDAarray(texture_fmt, *data.shape[::-1])
+    res_desc = texture.ResourceDescriptor(
+        runtime.cudaResourceTypeArray, cuArr=array)
+    tex_desc = texture.TextureDescriptor(
+        (address_mode, ) * data.ndim, filter_mode, read_mode)
+    tex_obj = texture.TextureObject(res_desc, tex_desc)
+    array.copy_from(data)
+
+    return tex_obj
+
+
+@cupy._util.memoize(for_each_device=True)
```

Comment:
I don't think this is needed as it's done internally in `ElementwiseKernel`.
```suggestion
```

Reply:
btw It'd be good to move the two `ElementwiseKernel` to the module level. They do not depend on the input parameters so there's no need to construct new kernel objects on the fly.

Reply:
There is now four elementwise kernels, because I am adding support for texture objects that are bound to linear memory instead of a CUDA array (and require `texNDfetch()`, instead of `textND()`). But I agree, and will move them to module level.

Reply:
That's nice! Any advantage for using linear memory? Is it easier to code or more flexible?

Reply:
Never tried linear memory textures before to be honest! If I understand correctly, they are a bit easier to code and slightly faster at random reading at the expense of losing spatial optimizations (Z-order curve) and hardware texture filtering (hence why they need `texNDfetch()` instead of `texND()`). I found some benchmark comparing linear and cuda array backed textures and (it seems cuda array should be generally faster](https://github.com/ricsonc/linear_vs_texture_memory_cuda).